### PR TITLE
ErrorLogger updated on ContractFactory, Provider and Signer. 

### PR DESCRIPTION
**Pull request by danielpavelicconsensys**
Final PR for ErrorLogger updates.

### CODE REVIEW

**Review by kalote — CHANGES_REQUESTED** [review]
one small modification

**Comment by salimtb** [comment]
i think we can make it better when we will move to Typescript by using generic types, here is a proposition but as the PR is in JS i'm wondering if we can create a ticket for that and implement it in the TS version: 
`function throwIfFalsy<T>(x: T | null | undefined, message: string, location: string): asserts x is T {
    if (!x) {
        throw new Error(message);
        throw new Error(
              errorLogger({
                location,
                message
              }),
            );
    }
}`

example of use here:

`throwIfFalsy(rpcUrl, ERROR_LOG.location.Provider_getProvider, ERROR_LOG.message.no_rpcURL)`

File: src/lib/Provider/Provider.js [hunk 1]
```diff
@@ -5,23 +5,43 @@
...
     return new ethers.providers.getDefaultProvider(rpcUrl);
   }
 
   static getInjectedProvider(injectedProvider) {
-    if (!injectedProvider) throw new Error('[Provider.getProvider] provider is missing!');
+    if (!injectedProvider) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.Provider_getInjectedProvider,
+          message: ERROR_LOG.message.no_privateKey,
```
[the remark this answers is not in the document]
Comment (danielpavelicconsensys) [reply src/lib/Provider/Provider.js]: changed, tx

**Review by salimtb — APPROVED** [review]

File: integration-test/erc721mintable.readContract.test-integration.js [hunk 2]
```diff
@@ -60,6 +60,7 @@ describe('E2E Test: Sdk (read)', () => {
       });
       const expectedTokenMetadata = {
         contract: '0x2a66707e4ffe929cf866bc048e54ce28f6b7275f',
+        attributes: [],
```
Comment (kalote) [inline integration-test/erc721mintable.readContract.test-integration.js:63]: FYI, this is linked to this => https://github.com/ConsenSys/nft-api-apps/issues/113

**Review by kalote — APPROVED** [review]

**Pull request merged**